### PR TITLE
Extensible PixelName API

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/0/1200049149472987/f
Tech Design URL: https://app.asana.com/0/715106103902962/1200043236683507/f
CC: 

**Description**:
This PR
* abstracts the `PixelName` as an interface so that it can be extended
* pixel definitions can thus be kept private to the gradle module that uses them

**Steps to test this PR**:
1. App should comile

(it is not extrictly necessary to perform any test as all issues should be detected at compile time)

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: statistics/src/main/java/com/duckduckgo/app/statistics/pixels/Pixel.kt [hunk 1]
```diff
@@ -18,22 +18,25 @@ package com.duckduckgo.app.statistics.pixels
...
 
-        FORGET_ALL_PRESSED_BROWSING("mf_bp"),
-        FORGET_ALL_PRESSED_TABSWITCHING("mf_tp"),
-        FORGET_ALL_EXECUTED("mf"),
-        FORGET_ALL_AUTO_RESTART("m_f_r"),
-        FORGET_ALL_AUTO_RESTART_WITH_INTENT("m_f_ri"),
+        COOKIE_DATABASE_NOT_FOUND("m_cdb_nf"),
+        COOKIE_DATABASE_OPEN_ERROR("m_cdb_oe"),
+        COOKIE_DATABASE_DELETE_ERROR("m_cdb_de"),
+        COOKIE_DATABASE_CORRUPTED_ERROR("m_cdb_ce"),
```
Comment (aitorvs) [inline statistics/src/main/java/com/duckduckgo/app/statistics/pixels/Pixel.kt:39]: These pixels are use in the `OfflinePixelSender` that lives in the `statistics` gradle module.
I think at some point we should also look into not having these pixels in the `statistics` gradle module. Ideally it should only contain the public API for the pixels, but none of the pixel names to be sent.
for now I just left these here

File: app/src/androidTest/java/com/duckduckgo/app/statistics/pixels/PixelNameTest.kt [hunk 2]
```diff
@@ -24,11 +25,15 @@ class PixelNameTest {
     @Test
     fun verifyNoDuplicatePixelNames() {
         val existingNames = mutableSetOf<String>()
-        Pixel.PixelName.values().forEach {
+        AppPixelName.values().forEach {
```
Comment (malmstein) [inline app/src/androidTest/java/com/duckduckgo/app/statistics/pixels/PixelNameTest.kt:28]: 👍

File: app/src/androidTest/java/com/duckduckgo/app/statistics/pixels/PixelNameTest.kt [hunk 3]
```diff
@@ -24,11 +25,15 @@ class PixelNameTest {
...
+            if (!existingNames.add(it.pixelName)) {
+                fail("Duplicate pixel name found: ${it.pixelName}")
```
Comment (malmstein) [inline app/src/androidTest/java/com/duckduckgo/app/statistics/pixels/PixelNameTest.kt:30]: Maybe be more specific towards what type of pixel is missing? Something like "Duplicate app / statistics pixel name found"

**Review by malmstein — APPROVED** [review]
small comment on the test logging, otherwise great work @aitorvs

**Pull request merged**